lib/pure/includes/osenv.nim

```diff
@@ -175,6 +181,32 @@ proc putEnv*(key, val: string) {.tags: [WriteEnvEffect].} =
       if c_putenv(environment[indx]) != 0'i32:
         raiseOSError(osLastError())
 
+proc unsetEnv*(key: string) {.tags: [WriteEnvEffect].} =
+  ## Removes the `environment variable`:idx: named `key`.
+  ## If an error occurs, `OSError` is raised.
+  ##
+  ## See also:ven
+  ## * `getEnv proc <#getEnv,string,string>`_
+  ## * `existsEnv proc <#existsEnv,string>`_
+  ## * `putEnv proc <#putEnv,string,string>`_
+  ## * `envPairs iterator <#envPairs.i>`_
+  when nimvm:
+    discard "built into the compiler"
+  else:
+    var indx = findEnvVar(key)
+    if indx < 0: return # Do nothing if the env var is not already set
+    let keyToRemove = environment[indx].split('=')[0] # "KEY=VAL" -> "KEY"; assumes that KEY does not have '=' char.
```

Comment:
To perhaps restate @Araq's point, you probably don't need to split on ALL the `=`s, right?

Reply:
yes, I don't need to split on ALL `=`. That's why I am taking in just `[0]`. I can set `maxsplit=1` or just use `find`.

@Araq So you need to replace the `split` call with `find` above?

Reply:
@Araq Actually, turns out that I didn't even need to use `find`. Please see the updated PR.